sm/mprv.h

```diff
@@ -0,0 +1,89 @@
+#pragma once
+#include <stdint.h>
+#include <stdbool.h>
+
+int copy1_from_sm(uintptr_t dst, const uint8_t *src);
```

Comment:
Why do we have separate functions for different lengths?

Reply:
I made these primitives so I would only have to write the minimum amount of assembly, since it's harder to read and maintain than C. The general variable-length loop is in `mprv.c`